src/packagedcode/opam.py

```diff
@@ -0,0 +1,249 @@
+    def get_package_root(cls, manifest_resource, codebase):
+        return manifest_resource.parent(codebase)
+
+    def repository_homepage_url(self, baseurl=default_web_baseurl):
+        if self.name:
+            return '{}/{}'.format(baseurl, self.name)
+
+
+def is_opam(location):
+    if location.endswith('.opam') or (filetype.is_file(location) and fileutils.file_name(location).lower() == 'opam'):
+        return True
+    return False
+
+
+def parse(location):
+    if not is_opam(location):
+        return
+
+    package_data = parse_opam(location)
+    return build_opam_package(package_data)
+
+
+def build_opam_package(package_data):
+    """
+    Return a Package object from a opam file or None.
+    """
+    package_dependencies = []
+    deps = package_data.get('depends') or []
+    for dep in deps:
+        package_dependencies.append(
+            models.DependentPackage(
+                purl=dep.purl,
+                requirement=dep.version,
+                scope='dependency',
+                is_runtime=True,
+                is_optional=False,
+                is_resolved=False,
+            )
+        )
+
+    name = package_data.get('name')
+    maintainers = package_data.get('maintainer') or []
+    authors = package_data.get('authors') or []
+    homepage_url = package_data.get('homepage')
+    vcs_url = package_data.get('dev-repo')
+    summary = package_data.get('synopsis')
+    description = package_data.get('description')
+    if summary:
+        description = summary
+    elif summary and description:
+        if len(summary) > len(description):
+            description = summary
+    
+    parties = []
+    for author in authors:
+        parties.append(
+            models.Party(
+                type=models.party_person,
+                name=author,
+                email=None,
+                role='author'
+            )
+        )
+    
+    for maintainer in maintainers:
+        parties.append(
+            models.Party(
+                type=models.party_person,
+                name=None,
+                email=maintainer,
+                role='email'
+            )
+        )
+
+    package = OpamPackage(
+        name=name or None,
+        vcs_url=vcs_url or None,
+        homepage_url=homepage_url,
+        description=description,
+        parties=parties,
+        dependencies=package_dependencies
+    )
+
+    return package
+
+
+@attr.s()
+class Opam(object):
+    name = attr.ib(default=None)
+    version = attr.ib(default=None)
+
+    @property
+    def purl(self):
+        return PackageURL(
+                    type='opam',
+                    name=self.name
+                ).to_string()
+
+
+# Regex expressions to parse file lines
+parse_file_line = re.compile(
+    r'(?P<key>^(.+?))'
+    r'\:\s*'
+    r'(?P<value>(.*))'
+).match
+
+parse_dep = re.compile(
+    r'^\s*\"'
+    r'(?P<name>[A-z0-9\-]*)'
+    r'\"\s*'
+    r'(?P<version>(.*))'
+).match
+
+
+def parse_opam(location):
+    """
```

Comment:
What is this file supposed to be?

Reply:
Ohh. I have added `opam data` in docstring

Reply:
I Think the docstring should be something this way:
```
Return a mapping of package data collected from the opam OCaml package manifest file at `location`.
```
that was the point I was making... you need to explain the arguments.

Reply:
:+1: